Framework/PythonInterface/mantid/plots/__init__.py

```diff
@@ -466,8 +467,15 @@ def replace_workspace_artists(self, workspace):
         except KeyError:
             return False
 
+        is_empty_list = []
         for workspace_artist in artist_info:
```

Comment:
This loop could be replaced with a generator, e.g:
```python
is_empty_list = [workspace_artist.replace_data(workspace) for workspace_artist in artist_info]
```

Reply:
I had copied this loop from two similar loops that also created an is_empty_list higher up in the same unit. I've changed all 3 instances now to use the list comprehension you suggested